### ISSUE TITLE
Set up structured database for the funding sources

### ISSUE THREAD

**Issue by npscience** [npscience/OpenCon-FundingDB#6]
Requirements:

* Easy to add suggestions to
* Requires approval for modifications?
* Easy to call up from website search

Data types will be informed by the user research survey ( #3 )

**Comment by npscience** [comment]
Could we use .json files in github? How is easiest to prepare the entry for these, e.g. a la Danielle and Max's template for the Data Rescue project: https://github.com/daniellecrobinson/Data-Rescue-PDX/blob/master/issue_template.md

**Comment by RaoOfPhysics** [comment]
Modifications/suggestions via PR? Would work. You can use CI to test, get the admin of the repo to approve all PRs etc.

**Comment by RaoOfPhysics** [comment]
Look at how the [homebrew cask](https://caskroom.github.io/) project works:
- Search database: https://caskroom.github.io/search
- PR submitted for changes to DB, and Travis CI checks the updates for certain parameters: https://travis-ci.org/caskroom/homebrew-cask/builds/226088578
- If CI passes, the PR is merged by admin

**Comment by npscience** [comment]
Joe suggested trying out the tech R2RC use for their speakers database as a rough prototype http://www.righttoresearch.org/resources/Speakersdatabase/

**Comment by npscience** [comment]
@RaoOfPhysics 

> Modifications/suggestions via PR? Would work. You can use CI to test, get the admin of the repo to approve all PRs etc.

Yes, this seems sensible. My hesitation is whether the intended audience (postdocs) would feel able to use github, or whether it would put them off. However, maybe postdocs just send us their spreadsheets, and github-initiated translate this into json files here.

**Comment by npscience** [comment]
@RaoOfPhysics 

> Look at how the homebrew cask project works

Thanks - this looks awesome. I like how easy the search is. I wonder how elastic the search matching could be - likely there will be lots of different ways of referring to a funding. But we could say 'who's the funder?' then work down from there.

**Comment by RaoOfPhysics** [comment]
>> Modifications/suggestions via PR? Would work. You can use CI to test, get the admin of the repo to approve all PRs etc.

> Yes, this seems sensible. My hesitation is whether the intended audience (postdocs) would feel able to use github, or whether it would put them off. However, maybe postdocs just send us their spreadsheets, and github-initiated translate this into json files here.

It depends on what information you're looking for. If they're uncomfortable with JSON, you could use a separate YAML file for each entry (overkill?); after all, that's essentially plaintext, and you could pre-populate it as a template. Alternatively, a [GitHub issues template](https://help.github.com/articles/creating-an-issue-template-for-your-repository/) with the relevant headers, so the postdocs can simple create an issue and you can take it from there?

Or ask them to send CSVs. :D

>> Look at how the homebrew cask project works

> Thanks - this looks awesome. I like how easy the search is. I wonder how elastic the search matching could be - likely there will be lots of different ways of referring to a funding. But we could say 'who's the funder?' then work down from there.

Makes sense! But I don't know anything about coding searches. So…

**Comment by npscience** [comment]
This tweet suggests a crowd based effort is beginning https://twitter.com/ahalltweetz/status/892042956768776192

The google sheet is at https://docs.google.com/spreadsheets/u/1/d/1X9sDHrerL6AdvWzXoIoSf8KSU5t6nLzFmSMTh9nRDcE/htmlview

**Comment by npscience** [comment]
Note there is also database of submitted grant proposals at ogrants.org
